File: tests/kotlin/tests-kotlin-gradle-junit5-allure3-selenium/src/test/java/helpers/WebDriverHolder.java
package helpers;

import org.openqa.selenium.WebDriver;

public final class WebDriverHolder {

    private static final ThreadLocal<WebDriver> DRIVER = new ThreadLocal<>();

    private WebDriverHolder() {
    }

    public static void set(WebDriver driver) {
        DRIVER.set(driver);
    }

    public static WebDriver get() {
        var driver = DRIVER.get();
        if (driver == null) {
            throw new IllegalStateException("WebDriver is not started");
        }
        return driver;
    }

    public static boolean has() {
        return DRIVER.get() != null;
    }

    public static void quit() {
        var driver = DRIVER.get();
        if (driver == null) {
            return;
        }
        try {
            driver.quit();
        } finally {
            DRIVER.remove();
        }
    }
}

File: tests/csharp/tests-csharp-nunit-restsharp-selenium/Helpers/WebDriverHolder.cs
using OpenQA.Selenium;

namespace Helpers;

public static class WebDriverHolder
{
    [ThreadStatic]
    private static IWebDriver? _driver;

    public static void Set(IWebDriver driver) => _driver = driver;

    public static IWebDriver Get() =>
        _driver ?? throw new InvalidOperationException("WebDriver is not started");

    public static bool Has => _driver != null;

    public static void Quit()
    {
        if (_driver == null)
        {
            return;
        }

        try
        {
            _driver.Quit();
        }
        finally
        {
            _driver = null;
        }
    }
}
